### PR TITLE
Setup repo and add fifo queue + clock

### PR DESCRIPTION
**Pull request by 0xekez**
This adds a FIFO queue package and the clock contract.

### CODE REVIEW

**Comment by 0xekez** [comment]
Some remaining TODO items:

1. Setup CI (interchaintest + schema generation + unit test).
2. Test clock.

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,29 @@
+Copyright 2022 Timewave
```
Comment (bekauz) [inline LICENSE:1]:
```suggestion
Copyright 2023 Timewave
```

File: packages/cw-fifo/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    /// elements are in the queue, returns None. O(1)
+    pub fn dequeue(&self, storage: &mut dyn Storage) -> StdResult<Option<T>> {
+        let Some((time, t)) = self
+            .0
+            .range(storage, None, None, cosmwasm_std::Order::Ascending)
+            .next()
+            .transpose()? else {
+		return Ok(None)
+	    };
+        self.0.remove(storage, time)?;
```
Comment (bekauz) [inline packages/cw-fifo/src/lib.rs:51]: so cool!

**Review by bekauz — APPROVED** [review]
looks great. love the FIFOQueue storage hack!

**Pull request merged**